janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/CustomAnalyzer.java

```diff
@@ -0,0 +1,119 @@
+    private static final String STANDARD_ANALYZER = org.apache.lucene.analysis.standard.StandardAnalyzer.class.getName();
+
+    private final String store;
+    private final KeyInformation.IndexRetriever informations;
+
+    private final Map<String, Analyzer> analyzers = new HashMap<>();
+
+    public CustomAnalyzer(String store, KeyInformation.IndexRetriever informations, ReuseStrategy fallbackStrategy) {
+        super(fallbackStrategy);
+        this.store = store;
+        this.informations = informations;
+    }
+
+    @Override
+    final protected Analyzer getWrappedAnalyzer(String fieldName) {
+        Mapping mapping = null;
+        String stringAnalyzerString = null;
+        String textAnalyzerString = null;
+
+
+        Parameter[] parameters = null;
+        Class<?> clazz = null;
+
+
+        KeyInformation keyInformation = informations.get(store, fieldName);
+        if (keyInformation != null) {
+            parameters = keyInformation.getParameters();
+            clazz = keyInformation.getDataType();
+        }
+
+        if (parameters != null) {
+            // if mapping isn't present in parameters, we use Mapping.DEFAULT
+            mapping = ParameterType.MAPPING.findParameter(parameters, Mapping.DEFAULT);
+            stringAnalyzerString = ParameterType.STRING_ANALYZER.findParameter(parameters, null);
+            textAnalyzerString = ParameterType.TEXT_ANALYZER.findParameter(parameters, null);
+        }
+      
+        String analyzerString = null;
+
+        if (clazz != null && String.class.isAssignableFrom(clazz) && mapping != null) {
+            // at the moment, we only try to support custom analyzers for string data.
+            // everything else falls through a StandardAnalyzer as was the case before
+            switch (mapping) {
+                case TEXTSTRING:
+                    throw new RuntimeException("TextString is an unsupported mapping for string data & custom analyzers");
+                case PREFIX_TREE:
+                    throw new RuntimeException("Prefix-tree is an unsupported mapping for string data & custom analyzers");
+                case STRING:
+                    analyzerString = stringAnalyzerString != null ? stringAnalyzerString : KEYWORD_ANALYZER;
+                    break;
+                case TEXT:
+                case DEFAULT:// TEXT
+                    analyzerString = textAnalyzerString;
+                    break;
+                default:
+                    throw new RuntimeException("Not supported");
+            }
+        }
+
+        // the default (& fall through) Analyzer is a StandardAnalyzer
+        if ((clazz == null || !String.class.isAssignableFrom(clazz)) || mapping == null || analyzerString == null) {
+            analyzerString = STANDARD_ANALYZER;
+        }
+
+        final String analyzerStringFinal = analyzerString;
+
+        Analyzer analyzer = analyzers.get(analyzerString);
```

Comment:
I did not do this change (but If you really want it, I'll do it) because that is the way I was doing it before david's review and he asked for this change (no lambda, less object creation, etc...)
I marked the place in his review where he asked for this

Reply:
Nope, good call. I'll keep in mind that the "computeIfAbsent" approach isn't preferred when I re-review.